### PR TITLE
[SAPBTPCFS-3288] update readme

### PR DESCRIPTION
**Pull request by I065450**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -451,11 +451,11 @@ stringData:
 ```
 ###### Custom Templates
 
-For additional flexibility, you can model your `Secret` resources according to your specific needs.
+For additional flexibility, you can model your `Secret` resources according to your specific needs.<br>
 To generate a custom-formatted `Secret`, use the `secretTemplate` attribute in the `ServiceBinding` spec.
-The value of the `secretTemplate` attribute must be a Go template, The generated output of the template should be in YAML format, representing the desired 'Secret'.
-The secret will be created with the data provided in the template and if `secretKey` and `secretRootKey` attributes are provided, they will be ignored.
-To include the default data in the secret (as provided by us), you should not include the `data` and `stringData` fields in your `secretTemplate`.
+The value of the `secretTemplate` attribute must be a Go template, The generated output of the template should be in YAML format, representing the desired 'Secret',
```
Comment (danielforsap) [inline README.md:456]: @I065450  proposal:

The value of the 'secretTemplate' attribute must be a Go template. It is mandatory to specify in the output format section of the template YAML as a format for the Secret.

`with . format: YAML`  - not sure how exactly it looks but something along these lines.

Note that you should also specify which attributes of your customized Secret resource should be base64 encoded, and which should be a plain string. 
For example:
s. apiVersion: v1
kind: Secret
metadata:
  name: my-secret
  namespace: default
data:
  username: {{ .Username | b64encode }} 
  password: {{ .Password | b64encode }} 
stringData:
  api_url: {{ .APIURL }}

If you do not specify this, the template will use default settings as defined by us for the attributes (from what I see, it is only stringData?)

Comment (danielforsap) [reply README.md:456]: *will use the encoding settings as provided by us.

Comment (danielforsap) [reply README.md:456]: the value of .... should be a YAML file of the Secret structure specified 
labels annotations - see the example 
The template should be in the YAML format with the structure of the Kubernetes Secret. Note that in the Secret Metadata, you can customize only its labels and annotations.

Comment (danielforsap) [reply README.md:456]: if a user adds secretKey or secretRootKey or nothing but specifies stringdata or data in the template, then the template takes precedence 
if they don't use data or stringdata in the template, then secretKey/secretRootKey and none are applied cause they only customized the metadata part of the Secret structure. 
format of stringdata will be one of the three (key-value pairs, credentials as JSON, credentials and service info as JSON)

Comment (danielforsap) [reply README.md:456]: provide explanation - bulleted list

File: README.md [hunk 2]
```diff
@@ -449,19 +449,17 @@ stringData:
         type: sample-instance-offering, // The service offering name
     }
 ```
-###### Custom Templates
+##### Custom Templates
```
Comment (avilupu) [inline README.md:452]: "Custom Secret Format"

u dont customize the template, the template is the *mean* to customize how the secret would look like.

File: README.md [hunk 3]
```diff
@@ -507,6 +508,60 @@ stringData:
   PASSWORD: ********
 ```
 
+###### Additional Information about Customization
+
+You can customize `Secret`'s `metadata` and `stringData` sections.
+
+- In the `metadata`, you can customize only its labels and annotations.
+- You can customize `stringData` or utilize one of the available formatting options as detailed in the [Formats of Service Binding Secrets](#formats-of-service-binding-secrets) section.
```
Comment (avilupu) [inline README.md:516]: it is not that clear

I would mention explicitly that if the string data is not customized then one of the "predefined" formats is used.

File: README.md [hunk 4]
```diff
@@ -507,6 +508,60 @@ stringData:
...
+
+**Important**:  If you customize `stringData`, your customization takes precedence over the pre-defined formats (if you parallely provided one of them).
```
Comment (avilupu) [inline README.md:518]: shall be before the 1st example.

File: README.md [hunk 5]
```diff
@@ -507,6 +508,60 @@ stringData:
...
+
```
Comment (avilupu) [inline README.md:519]: In general, I would change the order...

<table>

You can customize `Secret`'s `metadata` and `stringData` sections.
- In the `metadata`, you can customize only its labels and annotations.

**Important**:  If you customize `stringData`, your customization takes precedence over the pre-defined formats (if you parallely provided one of them).

<sample>
If needed you can utilize one of the available formatting options as detailed in the [Formats of Service Binding Secrets](#formats-of-service-binding-secrets) section, and customize only the metadata section.
E.g., adding labels or annotations to the secret, without changing the pre-defined format of the secret.

Example of using both the pre-defined formats and customization of the Metadata section:

<2nd sample>

File: README.md (hunk 3 above)
Comment (danielforsap) [reply README.md:516]: @avilupu but it isn't "used" unless a user specified one of the keys.
Technically, the one that is used automatically is just key-value pairs.

File: README.md [hunk 6]
```diff
@@ -449,19 +449,17 @@ stringData:
...
 
-Refer to [Go Templates](https://pkg.go.dev/text/template) for more details.
+This attribute expects a Go template as its value (for more information, see [Go Templates](https://pkg.go.dev/text/template)).<br>
 
-**Note** 
-If `secretTemplate` is used, both `secretKey` and `secretRootKey` attributes are ignored when provided.
-
-The template can use the following data: 
+**Note**<br>
+Make sure that the template is in YAML format, and that its structure is of a Kubernetes `Secret`.
```
Comment (I065450) [inline README.md:461]: I don't think it should be in a note.

Comment (danielforsap) [reply README.md:461]: @I065450  It is kind of important to emphasize, we can write Important.

I would still say something about which things are customizable first and then say that in spite of this flexibility we give, some things are לא ניתן להתפשר עליהם
I mean, not exactly those words, but you get what I mean.
Then we say that it has to be a YAML doc and with the Kubernetes Secret structure.

**Review by danielforsap — APPROVED** [review]

**Pull request merged**